source/dofs/dof_handler.cc

```diff
@@ -312,6 +312,8 @@ namespace internal
                 dof_handler.get_fe().dofs_per_cell,
               numbers::invalid_dof_index);
           }
+        dof_handler.non_local_dofs.resize(
+          dof_handler.get_fe().n_non_local_dofs(), numbers::invalid_dof_index);
```

Comment:
does this compile? I think the function should be called as in `dof_handler.get_fe().n_non_local_dofs_PER_CELL()` (because that's the name of the function in class `FiniteElement`). But this raises questions like I've already mentioned in the other PRs and issues: This suggests that these shape functions are not "per cell", but in fact global.

Reply:
This actually returns the total number on `non_local_dofs`